javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalPreservingPrinter.java

```diff
@@ -629,6 +636,12 @@ private static NodeText interpret(Node node, CsmElement csm, NodeText nodeText)
                         indentation.add(new TokenTextElement(SPACE, " "));
                     }
                 }
+                else if (indexCurrentElement > 0 && TokenTypes.isEndOfLineToken(IF)
```

Comment:
Thank you for your contribution, but this section of code is too poorly documented for us to understand its intent and context.

Reply:
OK thanks for your reminding and I will review this part of code soon